kernel/shared/src/main/scala/cats/effect/kernel/GenSpawn.scala

```diff
@@ -21,19 +21,276 @@ import cats.data.{EitherT, Ior, IorT, Kleisli, OptionT, WriterT}
+ * a fiber can request the abnormal termination of another fiber by calling
+ * [[Fiber!.cancel]].
+ *
+ * The cancellation model dictates that external cancellation behaves
+ * identically to self-cancellation. To guarantee consistent behavior between
+ * the two, the following semantics are shared:
+ *
+ *   1. Masking: if a fiber is cancelled while it is masked, cancellation is
+ *      suppressed until it reaches a completely unmasked state. See
+ *      [[MonadCancel]] documentation for more details.
+ *   1. Backpressure: [[Fiber!.cancel cancel]] semantically blocks all callers
+ *      until finalization is complete.
+ *   1. Idempotency: once a fiber's cancellation has been requested, subsequent
+ *      cancellations have no effect on cancellation status.
+ *   1. Terminal: Cancellation of a fiber that has terminated immediately
+ *      returns.
+ *
+ * External cancellation contrasts with self-cancellation in one aspect: the
+ * former may require synchronization between multiple threads to communicate
+ * a cancellation request. As a result, cancellation may not be immediately
+ * observed by a fiber. Implementations are free to decide how and when this
+ * synchronization takes place.
+ *
+ * ==Cancellation safety==
+ *
+ * A function or effect is considered to be cancellation-safe if it can be run
+ * in the absence of masking without violating effectful lifecycles or leaking
+ * resources. These functions require extra attention and care from users to
+ * ensure safe usage.
+ *
+ * [[start]] and [[racePair]] are both considered to be cancellation-unsafe
+ * effects because they return a [[Fiber]], which is a resource that has a
+ * lifecycle.
+ *
+ * {{{
+ *
+ *   // Start a fiber that continuously prints "A".
+ *   // After 10 seconds, cancel the fiber.
+ *   F.start(F.delay(println("A")).foreverM).flatMap { fiber =>
+ *     F.sleep(10.seconds) *> fiber.cancel
+ *   }
+ *
+ * }}}
+ *
+ * In the above example, imagine the spawning fiber is cancelled after it
+ * starts the printing fiber, but before the latter is cancelled. In this
+ * situation, the printing fiber is cancelled and will continue executing
+ * forever, contending with other fibers for system resources. This is known
+ * as a fiber leak, which is similar to a thread leak.
+ *
+ * Resource leaks like this are unfavorable when writing applications. In
+ * the case of [[start]] and [[racePair]], it is recommended against using
+ * these methods in favor of [[background]] and [[race]] respectively.
+ *
+ * The following example depicts a safer version of the `start` example
+ * above:
+ *
+ * {{{
+ *
+ *   // Starts a fiber that continously prints "A".
+ *   // After 10 seconds, the resource scope exits so the fiber is cancelled.
+ *   F.background(F.delay(println("A")).foreverM).use { _ =>
+ *     F.sleep(10.seconds)
+ *   }
+ *
+ * }}}
+ *
+ * ==Scheduling==
+ *
+ * Fibers are commonly referred to as *lightweight threads* or *green threads*.
```

Comment:
I do not believe asterisks italicize. Instead, I think we have to use `''`:

```suggestion
 * Fibers are commonly referred to as ''lightweight threads'' or ''green threads''.
```